lib/full-site-editing/edit-site-export.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * REST endpoint for exporting the contents of the Edit Site Page editor.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Output a ZIP file with an export of the current templates
+ * and template parts from the site editor, and close the connection.
+ */
+function gutenberg_edit_site_export() {
+	// Create ZIP file and directories.
+	$filename = tempnam( get_temp_dir(), 'edit-site-export' );
+	if ( ! class_exists( 'ZipArchive' ) ) {
+		return new WP_Error( 'Zip Export not supported.' );
+	}
+	$zip = new ZipArchive();
+	$zip->open( $filename, ZipArchive::OVERWRITE );
+	$zip->addEmptyDir( 'theme' );
+	$zip->addEmptyDir( 'theme/block-templates' );
+	$zip->addEmptyDir( 'theme/block-template-parts' );
+
+	// Load templates into the zip file.
+	$templates = gutenberg_get_block_templates( array( 'theme' => wp_get_theme()->get_stylesheet() ) );
+	foreach ( $templates as $template ) {
+		$zip->addFromString(
+			'theme/block-templates/' . $template->slug . '.html',
+			$template->content
+		);
+	}
+
+	// Load template parts into the zip file.
+	$template_parts = gutenberg_get_block_templates( array( 'theme' => wp_get_theme()->get_stylesheet() ), 'wp_template_part' );
+	foreach ( $template_parts as $template_part ) {
+		$zip->addFromString(
+			'theme/block-template-parts/' . $template_part->slug . '.html',
+			$template_part->content
+		);
+	}
+
+	// Send back the ZIP file.
+	$zip->close();
+	header( 'Content-Type: application/zip' );
+	header( 'Content-Disposition: attachment; filename=edit-site-export.zip' );
```

Comment:
Not added here, but let's maybe include the theme name in the filename (maybe in a follow-up). Something like

```suggestion
	header( "Content-Disposition: attachment; filename=$theme-export.zip" );
```

Reply:
Going to leave the export improvements separate, just to contain the scope of the PR.